File: dotnet/01-commands/explain/LifeGame/Game.cs

namespace LifeGame
{
    internal class Game
    {
        private Grid _grid;
        public bool IsFinished { get; private set; }

        public Game(Grid grid)
        {
            _grid = grid;
            IsFinished = false;
        }

        internal void Run()
        {
            Grid nextGrid = _grid.NextState();

            if (_grid.IsEqual(nextGrid))
            {
                IsFinished = true;
            }
            _grid = nextGrid;
        }

        internal bool GridIsEmpty()
        {
            return _grid.IsEmpty;
        }
    }
}

File: java/01-commands/explain/sourcecode/src/main/java/com/training/app/Game.java
package com.training.app;

public class Game {
    private Grid _grid;
    private boolean isFinished;

    public Game(Grid grid) {
        _grid = grid;
        isFinished = false;
    }

    public void run() {
        Grid nextGrid = _grid.nextState();

        if (_grid.isEqual(nextGrid)) {
            isFinished = true;
        }
        _grid = nextGrid;
    }

    public boolean gridIsEmpty() {
        return _grid.isEmpty();
    }

    public boolean isFinished() {
        return isFinished;
    }
}
